src/lepton/jpgcoder.cc

```diff
@@ -464,7 +464,12 @@ std::unique_ptr<BaseEncoder> g_encoder;
 
 std::unique_ptr<BaseDecoder> g_reference_to_free;
 ServiceInfo g_socketserve_info;
+
+// Global and per-file settings, allowing m/t processing of single file using worker threads pool.
+// Per-file setting is reinitialized from the global one in prepare_for_new_file()
+bool option_threaded = true;
```

Comment:
Why is it needed?

Reply:
Original Lepton code initializes g_threaded from options, and then can modify it based on processed file (i.e. disable m/threading when file is small). So, processing multiple files requires to reset g_threaded back to value computed from options (option_threaded).